test_dirs/25_synth/configlog.txt
__name__ = config 
__doc__ = None 
__package__ =  
__loader__ = <_frozen_importlib_external.SourceFileLoader object at 0x00000201CFDC6880> 
__spec__ = ModuleSpec(name='config', loader=<_frozen_importlib_external.SourceFileLoader object at 0x00000201CFDC6880>, origin='D:\\Nextcloud_old\\pythoncode\\prewhitening_new\\config.py') 
__file__ = D:\Nextcloud_old\pythoncode\prewhitening_new\config.py 
__cached__ = D:\Nextcloud_old\pythoncode\prewhitening_new\__pycache__\config.cpython-39.pyc 
__builtins__ = {'__name__': 'builtins', '__doc__': "Built-in functions, exceptions, and other objects.\n\nNoteworthy: None is the `nil' object; Ellipsis represents `...' in slices.", '__package__': '', '__loader__': <class '_frozen_importlib.BuiltinImporter'>, '__spec__': ModuleSpec(name='builtins', loader=<class '_frozen_importlib.BuiltinImporter'>, origin='built-in'), '__build_class__': <built-in function __build_class__>, '__import__': <built-in function __import__>, 'abs': <built-in function abs>, 'all': <built-in function all>, 'any': <built-in function any>, 'ascii': <built-in function ascii>, 'bin': <built-in function bin>, 'breakpoint': <built-in function breakpoint>, 'callable': <built-in function callable>, 'chr': <built-in function chr>, 'compile': <built-in function compile>, 'delattr': <built-in function delattr>, 'dir': <built-in function dir>, 'divmod': <built-in function divmod>, 'eval': <built-in function eval>, 'exec': <built-in function exec>, 'format': <built-in function format>, 'getattr': <built-in function getattr>, 'globals': <built-in function globals>, 'hasattr': <built-in function hasattr>, 'hash': <built-in function hash>, 'hex': <built-in function hex>, 'id': <built-in function id>, 'input': <built-in function input>, 'isinstance': <built-in function isinstance>, 'issubclass': <built-in function issubclass>, 'iter': <built-in function iter>, 'len': <built-in function len>, 'locals': <built-in function locals>, 'max': <built-in function max>, 'min': <built-in function min>, 'next': <built-in function next>, 'oct': <built-in function oct>, 'ord': <built-in function ord>, 'pow': <built-in function pow>, 'print': <built-in function print>, 'repr': <built-in function repr>, 'round': <built-in function round>, 'setattr': <built-in function setattr>, 'sorted': <built-in function sorted>, 'sum': <built-in function sum>, 'vars': <built-in function vars>, 'None': None, 'Ellipsis': Ellipsis, 'NotImplemented': NotImplemented, 'False': False, 'True': True, 'bool': <class 'bool'>, 'memoryview': <class 'memoryview'>, 'bytearray': <class 'bytearray'>, 'bytes': <class 'bytes'>, 'classmethod': <class 'classmethod'>, 'complex': <class 'complex'>, 'dict': <class 'dict'>, 'enumerate': <class 'enumerate'>, 'filter': <class 'filter'>, 'float': <class 'float'>, 'frozenset': <class 'frozenset'>, 'property': <class 'property'>, 'int': <class 'int'>, 'list': <class 'list'>, 'map': <class 'map'>, 'object': <class 'object'>, 'range': <class 'range'>, 'reversed': <class 'reversed'>, 'set': <class 'set'>, 'slice': <class 'slice'>, 'staticmethod': <class 'staticmethod'>, 'str': <class 'str'>, 'super': <class 'super'>, 'tuple': <class 'tuple'>, 'type': <class 'type'>, 'zip': <class 'zip'>, '__debug__': True, 'BaseException': <class 'BaseException'>, 'Exception': <class 'Exception'>, 'TypeError': <class 'TypeError'>, 'StopAsyncIteration': <class 'StopAsyncIteration'>, 'StopIteration': <class 'StopIteration'>, 'GeneratorExit': <class 'GeneratorExit'>, 'SystemExit': <class 'SystemExit'>, 'KeyboardInterrupt': <class 'KeyboardInterrupt'>, 'ImportError': <class 'ImportError'>, 'ModuleNotFoundError': <class 'ModuleNotFoundError'>, 'OSError': <class 'OSError'>, 'EnvironmentError': <class 'OSError'>, 'IOError': <class 'OSError'>, 'WindowsError': <class 'OSError'>, 'EOFError': <class 'EOFError'>, 'RuntimeError': <class 'RuntimeError'>, 'RecursionError': <class 'RecursionError'>, 'NotImplementedError': <class 'NotImplementedError'>, 'NameError': <class 'NameError'>, 'UnboundLocalError': <class 'UnboundLocalError'>, 'AttributeError': <class 'AttributeError'>, 'SyntaxError': <class 'SyntaxError'>, 'IndentationError': <class 'IndentationError'>, 'TabError': <class 'TabError'>, 'LookupError': <class 'LookupError'>, 'IndexError': <class 'IndexError'>, 'KeyError': <class 'KeyError'>, 'ValueError': <class 'ValueError'>, 'UnicodeError': <class 'UnicodeError'>, 'UnicodeEncodeError': <class 'UnicodeEncodeError'>, 'UnicodeDecodeError': <class 'UnicodeDecodeError'>, 'UnicodeTranslateError': <class 'UnicodeTranslateError'>, 'AssertionError': <class 'AssertionError'>, 'ArithmeticError': <class 'ArithmeticError'>, 'FloatingPointError': <class 'FloatingPointError'>, 'OverflowError': <class 'OverflowError'>, 'ZeroDivisionError': <class 'ZeroDivisionError'>, 'SystemError': <class 'SystemError'>, 'ReferenceError': <class 'ReferenceError'>, 'MemoryError': <class 'MemoryError'>, 'BufferError': <class 'BufferError'>, 'Warning': <class 'Warning'>, 'UserWarning': <class 'UserWarning'>, 'DeprecationWarning': <class 'DeprecationWarning'>, 'PendingDeprecationWarning': <class 'PendingDeprecationWarning'>, 'SyntaxWarning': <class 'SyntaxWarning'>, 'RuntimeWarning': <class 'RuntimeWarning'>, 'FutureWarning': <class 'FutureWarning'>, 'ImportWarning': <class 'ImportWarning'>, 'UnicodeWarning': <class 'UnicodeWarning'>, 'BytesWarning': <class 'BytesWarning'>, 'ResourceWarning': <class 'ResourceWarning'>, 'ConnectionError': <class 'ConnectionError'>, 'BlockingIOError': <class 'BlockingIOError'>, 'BrokenPipeError': <class 'BrokenPipeError'>, 'ChildProcessError': <class 'ChildProcessError'>, 'ConnectionAbortedError': <class 'ConnectionAbortedError'>, 'ConnectionRefusedError': <class 'ConnectionRefusedError'>, 'ConnectionResetError': <class 'ConnectionResetError'>, 'FileExistsError': <class 'FileExistsError'>, 'FileNotFoundError': <class 'FileNotFoundError'>, 'IsADirectoryError': <class 'IsADirectoryError'>, 'NotADirectoryError': <class 'NotADirectoryError'>, 'InterruptedError': <class 'InterruptedError'>, 'PermissionError': <class 'PermissionError'>, 'ProcessLookupError': <class 'ProcessLookupError'>, 'TimeoutError': <class 'TimeoutError'>, 'open': <built-in function open>, 'quit': Use quit() or Ctrl-Z plus Return to exit, 'exit': Use exit() or Ctrl-Z plus Return to exit, 'copyright': Copyright (c) 2001-2021 Python Software Foundation.
All Rights Reserved.

Copyright (c) 2000 BeOpen.com.
All Rights Reserved.

Copyright (c) 1995-2001 Corporation for National Research Initiatives.
All Rights Reserved.

Copyright (c) 1991-1995 Stichting Mathematisch Centrum, Amsterdam.
All Rights Reserved., 'credits':     Thanks to CWI, CNRI, BeOpen.com, Zope Corporation and a cast of thousands
    for supporting Python development.  See www.python.org for more information., 'license': Type license() to see the full license text, 'help': Type help() for interactive help, or help(object) for help about object.} 
version = 1.0 
working_dir = D:/Nextcloud_old/pythoncode/prewhitening_new/test_dirs/25_synth 
target_file = 25_f_synthdata.dat 
cols = [0, 1, 2] 
dtype = mmag 
n_f = 25 
target_dtype = mmag 
strip_gaps = True 
gap_size = 0.5 
strip_gaps_points = 50 
log_fname = log.txt 
quiet = False 
preprocessed_lightcurve_fname = inp_lc.dat 
frequencies_fname = frequencies.csv 
fixed_freqamp_single = False 
fixed_freq_single = False 
fixed_none_single = True 
fixed_freqamp_multi = False 
fixed_freq_multi = True 
fixed_none_multi = True 
analytical_jacobians = True 
multi_fit_type = anneal 
ann_bounds_method = relative 
ann_relfrac_freq = 0.1 
ann_relfrac_amp = 0.2 
ann_relfrac_phase = 0.2 
ann_zp_bounds_lower = -10 
ann_zp_bounds_upper = 10 
periodograms_lowerbound = resolution 
periodograms_upperbound = 20 
figure_subdir = /figures 
iterative_subdir = /figures_iterative 
plot_iterative_lcs = True 
plot_iterative_pgs = True 
frequencies_da_filename = frequencies_da.csv 


test_dirs/25_synth - Copy/configlog.txt
__name__ = config 
__doc__ = None 
__package__ =  
__loader__ = <_frozen_importlib_external.SourceFileLoader object at 0x00000168C8446880> 
__spec__ = ModuleSpec(name='config', loader=<_frozen_importlib_external.SourceFileLoader object at 0x00000168C8446880>, origin='D:\\Nextcloud_old\\pythoncode\\prewhitening_new\\config.py') 
__file__ = D:\Nextcloud_old\pythoncode\prewhitening_new\config.py 
__cached__ = D:\Nextcloud_old\pythoncode\prewhitening_new\__pycache__\config.cpython-39.pyc 
__builtins__ = {'__name__': 'builtins', '__doc__': "Built-in functions, exceptions, and other objects.\n\nNoteworthy: None is the `nil' object; Ellipsis represents `...' in slices.", '__package__': '', '__loader__': <class '_frozen_importlib.BuiltinImporter'>, '__spec__': ModuleSpec(name='builtins', loader=<class '_frozen_importlib.BuiltinImporter'>, origin='built-in'), '__build_class__': <built-in function __build_class__>, '__import__': <built-in function __import__>, 'abs': <built-in function abs>, 'all': <built-in function all>, 'any': <built-in function any>, 'ascii': <built-in function ascii>, 'bin': <built-in function bin>, 'breakpoint': <built-in function breakpoint>, 'callable': <built-in function callable>, 'chr': <built-in function chr>, 'compile': <built-in function compile>, 'delattr': <built-in function delattr>, 'dir': <built-in function dir>, 'divmod': <built-in function divmod>, 'eval': <built-in function eval>, 'exec': <built-in function exec>, 'format': <built-in function format>, 'getattr': <built-in function getattr>, 'globals': <built-in function globals>, 'hasattr': <built-in function hasattr>, 'hash': <built-in function hash>, 'hex': <built-in function hex>, 'id': <built-in function id>, 'input': <built-in function input>, 'isinstance': <built-in function isinstance>, 'issubclass': <built-in function issubclass>, 'iter': <built-in function iter>, 'len': <built-in function len>, 'locals': <built-in function locals>, 'max': <built-in function max>, 'min': <built-in function min>, 'next': <built-in function next>, 'oct': <built-in function oct>, 'ord': <built-in function ord>, 'pow': <built-in function pow>, 'print': <built-in function print>, 'repr': <built-in function repr>, 'round': <built-in function round>, 'setattr': <built-in function setattr>, 'sorted': <built-in function sorted>, 'sum': <built-in function sum>, 'vars': <built-in function vars>, 'None': None, 'Ellipsis': Ellipsis, 'NotImplemented': NotImplemented, 'False': False, 'True': True, 'bool': <class 'bool'>, 'memoryview': <class 'memoryview'>, 'bytearray': <class 'bytearray'>, 'bytes': <class 'bytes'>, 'classmethod': <class 'classmethod'>, 'complex': <class 'complex'>, 'dict': <class 'dict'>, 'enumerate': <class 'enumerate'>, 'filter': <class 'filter'>, 'float': <class 'float'>, 'frozenset': <class 'frozenset'>, 'property': <class 'property'>, 'int': <class 'int'>, 'list': <class 'list'>, 'map': <class 'map'>, 'object': <class 'object'>, 'range': <class 'range'>, 'reversed': <class 'reversed'>, 'set': <class 'set'>, 'slice': <class 'slice'>, 'staticmethod': <class 'staticmethod'>, 'str': <class 'str'>, 'super': <class 'super'>, 'tuple': <class 'tuple'>, 'type': <class 'type'>, 'zip': <class 'zip'>, '__debug__': True, 'BaseException': <class 'BaseException'>, 'Exception': <class 'Exception'>, 'TypeError': <class 'TypeError'>, 'StopAsyncIteration': <class 'StopAsyncIteration'>, 'StopIteration': <class 'StopIteration'>, 'GeneratorExit': <class 'GeneratorExit'>, 'SystemExit': <class 'SystemExit'>, 'KeyboardInterrupt': <class 'KeyboardInterrupt'>, 'ImportError': <class 'ImportError'>, 'ModuleNotFoundError': <class 'ModuleNotFoundError'>, 'OSError': <class 'OSError'>, 'EnvironmentError': <class 'OSError'>, 'IOError': <class 'OSError'>, 'WindowsError': <class 'OSError'>, 'EOFError': <class 'EOFError'>, 'RuntimeError': <class 'RuntimeError'>, 'RecursionError': <class 'RecursionError'>, 'NotImplementedError': <class 'NotImplementedError'>, 'NameError': <class 'NameError'>, 'UnboundLocalError': <class 'UnboundLocalError'>, 'AttributeError': <class 'AttributeError'>, 'SyntaxError': <class 'SyntaxError'>, 'IndentationError': <class 'IndentationError'>, 'TabError': <class 'TabError'>, 'LookupError': <class 'LookupError'>, 'IndexError': <class 'IndexError'>, 'KeyError': <class 'KeyError'>, 'ValueError': <class 'ValueError'>, 'UnicodeError': <class 'UnicodeError'>, 'UnicodeEncodeError': <class 'UnicodeEncodeError'>, 'UnicodeDecodeError': <class 'UnicodeDecodeError'>, 'UnicodeTranslateError': <class 'UnicodeTranslateError'>, 'AssertionError': <class 'AssertionError'>, 'ArithmeticError': <class 'ArithmeticError'>, 'FloatingPointError': <class 'FloatingPointError'>, 'OverflowError': <class 'OverflowError'>, 'ZeroDivisionError': <class 'ZeroDivisionError'>, 'SystemError': <class 'SystemError'>, 'ReferenceError': <class 'ReferenceError'>, 'MemoryError': <class 'MemoryError'>, 'BufferError': <class 'BufferError'>, 'Warning': <class 'Warning'>, 'UserWarning': <class 'UserWarning'>, 'DeprecationWarning': <class 'DeprecationWarning'>, 'PendingDeprecationWarning': <class 'PendingDeprecationWarning'>, 'SyntaxWarning': <class 'SyntaxWarning'>, 'RuntimeWarning': <class 'RuntimeWarning'>, 'FutureWarning': <class 'FutureWarning'>, 'ImportWarning': <class 'ImportWarning'>, 'UnicodeWarning': <class 'UnicodeWarning'>, 'BytesWarning': <class 'BytesWarning'>, 'ResourceWarning': <class 'ResourceWarning'>, 'ConnectionError': <class 'ConnectionError'>, 'BlockingIOError': <class 'BlockingIOError'>, 'BrokenPipeError': <class 'BrokenPipeError'>, 'ChildProcessError': <class 'ChildProcessError'>, 'ConnectionAbortedError': <class 'ConnectionAbortedError'>, 'ConnectionRefusedError': <class 'ConnectionRefusedError'>, 'ConnectionResetError': <class 'ConnectionResetError'>, 'FileExistsError': <class 'FileExistsError'>, 'FileNotFoundError': <class 'FileNotFoundError'>, 'IsADirectoryError': <class 'IsADirectoryError'>, 'NotADirectoryError': <class 'NotADirectoryError'>, 'InterruptedError': <class 'InterruptedError'>, 'PermissionError': <class 'PermissionError'>, 'ProcessLookupError': <class 'ProcessLookupError'>, 'TimeoutError': <class 'TimeoutError'>, 'open': <built-in function open>, 'quit': Use quit() or Ctrl-Z plus Return to exit, 'exit': Use exit() or Ctrl-Z plus Return to exit, 'copyright': Copyright (c) 2001-2021 Python Software Foundation.
All Rights Reserved.

Copyright (c) 2000 BeOpen.com.
All Rights Reserved.

Copyright (c) 1995-2001 Corporation for National Research Initiatives.
All Rights Reserved.

Copyright (c) 1991-1995 Stichting Mathematisch Centrum, Amsterdam.
All Rights Reserved., 'credits':     Thanks to CWI, CNRI, BeOpen.com, Zope Corporation and a cast of thousands
    for supporting Python development.  See www.python.org for more information., 'license': Type license() to see the full license text, 'help': Type help() for interactive help, or help(object) for help about object.} 
version = 1.0 
working_dir = D:/Nextcloud_old/pythoncode/prewhitening_new/test_dirs/25_synth 
target_file = 25_f_synthdata.dat 
cols = [0, 1, 2] 
dtype = mmag 
n_f = 25 
target_dtype = mmag 
strip_gaps = True 
gap_size = 0.5 
strip_gaps_points = 50 
log_fname = log.txt 
quiet = False 
preprocessed_lightcurve_fname = inp_lc.dat 
frequencies_fname = frequencies.csv 
fixed_freqamp_single = False 
fixed_freq_single = False 
fixed_none_single = True 
fixed_freqamp_multi = False 
fixed_freq_multi = True 
fixed_none_multi = True 
analytical_jacobians = True 
multi_fit_type = anneal 
ann_bounds_method = relative 
ann_relfrac_freq = 0.1 
ann_relfrac_amp = 0.2 
ann_relfrac_phase = 0.2 
ann_zp_bounds_lower = -10 
ann_zp_bounds_upper = 10 
periodograms_lowerbound = resolution 
periodograms_upperbound = 20 
figure_subdir = /figures 
iterative_subdir = /figures_iterative 
plot_iterative_lcs = True 
plot_iterative_pgs = True 


config.py
import os
"""
Instructions for use:
1) Set everything under main - recommend making a directory and putting the timeseries file in it, then setting
the directory and target file appropriately.
2) If the data file is processed already, just set the data types to match (dtype and target_dtype), and set strip_gaps
to False.
3) Set other options - some are self-explanatory and others are less. All are poorly documented at the moment.
4) Run main.py
"""
# ========== Main/Most relevant ==========
version = "1.1"
working_dir = f"C:/Users/Erik/main/projects/plaskett_photometry/hd47129_extractions_final/16cbv_testing"
target_file = "HD47129_squaremask_hard_16CBV.txt"
cols = [0, 1, 2]
dtype = "flux"
n_f = 20  # number of frequencies to extract

averaging_bin_width = 2
cutoff_iteration = 6
cutoff_sig = 3

multi_fit_type = "lm"  # "anneal", "lm", "scipy". Sets fitting engine to use.
residual_model_generation = "mf" # can be sf, mf. Controls which model is used to generate residual periodogram

clean_existing = True

# ========== Preprocessing ==========
target_dtype = "mmag"  # allowed values: mag, mmag, flux (only if data originally in flux)
strip_gaps = True
gap_size = 0.5  # days
strip_gaps_points = 50  # on either side of each gap, and at the start and end
# ========== Logging ==========
log_fname = "log.txt"
save_freq_logs = True
freqlog_folder = "freq_logs"

# ========== Console ==========
quiet = False

# ========== Output targets ==========
preprocessed_lightcurve_fname = "inp_lc.dat"
frequencies_fname = "frequencies.csv"

# ========== Frequency Selection ==========
# TODO: implement LOPOLY fit for freq selection+rejection by significance
freq_selection_method = "highest" # can be highest, or averaged


# ========== Fitting ==========
fixed_freqamp_single = False
fixed_freq_single = False
fixed_none_single = True

phase_fit_rejection_criterion = 0.1

fixed_freqamp_multi = False
fixed_freq_multi = True
fixed_none_multi = True
# ========== LM ==========
# freq bounds use coefficients
lm_freq_bounds = True
lm_freq_bounds_lower_coef = 0.8
lm_freq_bounds_upper_coef = 1.2
# amp bounds use coefficients
lm_amp_bounds = True
lm_amp_bounds_lower_coef = 0.8
lm_amp_bounds_upper_coef = 1.2
# phase bounds set explicitly
lm_phase_bounds = True
lm_phase_bounds_lower = -100
lm_phase_bounds_upper = 100
# ========== Annealing ==========
ann_bounds_method = "relative"  # "explicit", "relative", "plusminus"
if ann_bounds_method=="explicit":
    ann_freq_bounds_lower = 0
    ann_freq_bounds_upper = 20
    ann_amp_bounds_lower = 0
    ann_amp_bounds_upper = 20
    ann_phase_bounds_lower = 0
    ann_phase_bounds_upper = 1
if ann_bounds_method == "relative":
    ann_relfrac_freq = 0.1
    ann_relfrac_amp = 0.2
    ann_relfrac_phase = 0.2
if ann_bounds_method == "plusminus":
    ann_pm_freq = 0.2
    ann_pm_amp = 1
    ann_pm_phase = 0.5
ann_zp_bounds_lower = -10
ann_zp_bounds_upper = 10
# ========== Fitting ==========
# ========== Periodograms ==========
periodograms_lowerbound = "resolution"  # can be set to resolution or explicit value in c/d
periodograms_upperbound = 20

# ========== Plots ==========

figure_subdir = "/figures"
iterative_subdir = "/figures_iterative"
plot_iterative_lcs = True
plot_iterative_pgs = True

# ========== Dual Annealing ==========
frequencies_da_filename = "frequencies_da.csv"
